pyscf/scf/hf.py

```diff
@@ -719,13 +721,38 @@ def fproj(mo):
         dm = lib.tag_array(dm, mo_coeff=mo_coeff[:,::-1], mo_occ=mo_occ)
     return dm
 
+def init_guess_by_vsap(mol, sapgrasp='small'):
+    '''Generate initial guess density matrix using the superposition of
+    atomic potentials (SAP) guess, doi:10.1021/acs.jctc.8b01089
```

Comment:
This implementation is based on the Gaussian fit presented in [doi:10.1063/5.0004046](https://doi.org/10.1063/5.0004046)